puffin/src/lib.rs

```diff
@@ -659,27 +697,85 @@ macro_rules! current_file_name {
     };
 }
 
-/// Removes long path prefix to focus on the last parts of the path (and the file name).
+/// Shortens a long `file!()` path to the essentials.
+///
+/// We want to keep it short for two reasons: readability, and bandwidth
 #[doc(hidden)]
-#[inline]
-pub fn short_file_name(name: &str) -> &str {
-    // TODO: "foo/bar/src/lib.rs" -> "bar/src/lib.rs"
+#[inline(never)]
+pub fn short_file_name(path: &'static str) -> String {
```

Comment:
Note that this _breaking_ API change has been published as non-semver-breaking `0.17.1` release?

Reply:
good catch, we should hide those functions with `#[doc(hidden)]` as don't see a reason to have them part of the public API.

hiding them in patch release is still a breaking change, though we are pre v1 and could give ourselves some flexibility and say they were considered internal details just exposed for the macros.

Reply:
We rely on them to reimplement the macros while also calling our own implementation with it next to `puffin`. Any way we could keep this around?

Reply:
I'd take it for that use case the `static` lifetime also doesn't work?

we could revert that specific change, it was more of style/semantics, not for any concrete benefit. didn't think of that these functions were part of the public interface.

@emilk we could revert it and publish another patch version without these to get back to the previous public API

Reply:
> I'd take it for that use case the `static` lifetime also doesn't work?

I think that's fine because we were calling the `puffin::current_file_name!()` macro. The expression it emits suddenly returns `String` while `ProfilerScope::new()` expects `&str` now.

Reply:
Maybe that also means we need to switch over to `std::once::Once` and `format!(...).leak()` here. Perhaps it is useful to see for us how much we can reuse, maybe I can actually call the `puffin` macro within our own macro.

Really the only thing we're doing extra in our macro is calling into https://crates.io/crates/superluminal-perf, from what I can tell (so that we don't have to annotate our functions with profiling markers for multiple tools).

Reply:
@emilk can you yank the breaking release? I'm going through dependency upgrades and keep forgetting to run `cargo update -p puffin --precise 0.17.0` to opt out of semver-breaking changes :sweat_smile: